test/gtest/uct/test_peer_failure.cc

```diff
@@ -263,21 +277,32 @@ UCS_TEST_SKIP_COND_P(test_uct_peer_failure, purge_failed_peer,
     send_recv_am(0);
     send_recv_am(1);
 
-    const size_t num_pend_sends = 3ul;
-    uct_pending_req_t reqs[num_pend_sends];
+    const ucs_time_t loop_end_limit = ucs::get_deadline();
+    const size_t num_pend_sends     = 64ul;
+    const uint64_t send_data        = 0;
+    std::vector<pending_send_request_t> reqs(num_pend_sends);
+
     {
         scoped_log_handler slh(wrap_errors_logger);
 
-        kill_receiver();
-
         ucs_status_t status;
         do {
-            status = uct_ep_am_short(ep0(), 0, 0, NULL, 0);
-        } while (status == UCS_OK);
+            status = uct_ep_am_short(ep0(), 0, 0, &send_data,
+                                     sizeof(send_data));
+        } while ((status == UCS_OK) && (ucs_get_time() < loop_end_limit));
+
+        if (status == UCS_OK) {
+            UCS_TEST_SKIP_R("unable to fill the UCT resources");
+        } else if (status != UCS_ERR_NO_RESOURCE) {
+            UCS_TEST_ABORT("AM Short failed with " << ucs_status_string(status));
+        }
+
+        kill_receiver();
 
         for (size_t i = 0; i < num_pend_sends; i ++) {
-            reqs[i].func = pending_cb;
-            EXPECT_EQ(uct_ep_pending_add(ep0(), &reqs[i], 0), UCS_OK);
+            reqs[i].ep       = ep0();
+            reqs[i].uct.func = pending_cb;
```

Comment:
in line 313: can we check how many requests were purged?

Reply:
it is already done in 317 line
rename the field `m_req_count` -> `m_req_purge_count`

Reply:
right, ok